src/numpy-stubs/_typing/_ufunc.pyi

```diff
@@ -380,6 +446,113 @@ class _Call21Bool(Protocol):
         **kwds: Unpack[_Kwargs3],
     ) -> np.bool | NDArray[np.bool]: ...
 
+@type_check_only
+class _Call21Logical(Protocol):
+    @overload  # (scalar, scalar, dtype: np.object_) -> np.object_
+    def __call__(
+        self,
+        x1: _NumberLike_co,
+        x2: _NumberLike_co,
+        /,
+        dtype: _DTypeLikeObject,
+        out: None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> np.object_: ...
+    @overload  # (scalar, scalar) -> bool
+    def __call__(
+        self,
+        x1: _NumberLike_co,
+        x2: _NumberLike_co,
+        /,
+        out: None = None,
+        dtype: DTypeLike | None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> np.bool: ...
+    @overload  # (array-like, array, dtype: np.object_) -> np.object_
+    def __call__(
+        self,
+        x1: _ArrayLikeNumber_co,
+        x2: _AnyArray,
+        /,
+        dtype: _DTypeLikeObject,
+        out: None = None,
+        **kwds: Unpack[_Kwargs3],
+    ) -> np.object_: ...
+    @overload  # (array-like, array, dtype: dtype[T]) -> Array[T]
+    def __call__(
+        self,
+        x1: _ArrayLikeNumber_co,
+        x2: _AnyArray,
```

Comment:
`_AnyArray` also accepts `NDArray[object_]`, which would result in a `NDArray[np.object_]`. You could avoid that with e.g.

```suggestion
        x2: NDArray[np.bool | np.number],
```

But perhaps it's cleaner to create an alias for it, e.g. `_CoNumberArray` (the `Co` implies that it's "**co**ercible to (`number`)", and therefore also includes `np.bool`)